src/foundations/shape/Shape.mixins.ts

```diff
@@ -0,0 +1,90 @@
+import { css } from 'styled-components';
+import { FlattenSimpleInterpolation } from 'styled-components/ts3.6';
+
+import { Utils } from '../../shared';
+
+import { SHAPE_ELEVATION_DEFINITIONS } from './Shape.constants';
+import { PApplyElevation, PApplyShape } from './Shape.props';
+import { TShapeVariant } from './Shape.types';
+
+/**
+ * apply a shape to a given element without the use of an additional wrapping Shape component.
+ *
+ * @param {(number|string)?} width
+ * @param {(number|string)?} height
+ * @param {TShapeBorderRadius} radius
+ * @returns {FlattenSimpleInterpolation}
+ */
+function applyShape({ width, height, radius }: PApplyShape): FlattenSimpleInterpolation {
+    // define the variant by checking for string
+    const variant: TShapeVariant = Utils.isString(radius) ? radius : 'rectangle';
+
+    // the circle variant has to have numerical width set for it to work
+    if (variant === 'circle' && !Utils.isNumber(width)) {
+        throw new TypeError(
+            'applyShape: When choosing `circle` as value for `radius` the width needs to be of type `number`'
+        );
+    }
+
+    const RADII: Partial<Record<TShapeVariant, string>> = {
+        rectangle: `${radius}px`,
```

Comment:
same here are we sure this will always be a valid value?

Reply:
it should